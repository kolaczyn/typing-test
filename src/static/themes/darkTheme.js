import { DARK_THEME } from '../../reducers/theme/actions';

const darkTheme = {
  palette: {
    box: '#222',
    border: '#1c1e21',
    background: '#303030',
    text: '#fff',

    green: '#55efc4',
    blue: '#00a1e4',
    red: '#e17055',
    yellow: '#e8e9a1',
    lightGray: 'bisque',
  },
  themeName: DARK_THEME,
};

export default darkTheme;
