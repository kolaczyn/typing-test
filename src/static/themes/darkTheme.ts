import Action from '../../reducers/theme/actions';
import ThemeType from './ThemeType';

// https://flatuicolors.com/palette/defo
const darkTheme: ThemeType = {
  palette: {
    box: '#2c3e50',
    border: '#27394B',
    background: '#34495e',
    text: '#ecf0f1',

    green: '#2ecc71',
    blue: '#3498db',
    red: '#f96454',
    yellow: '#f1c40f',
    lightGray: '#bdc3c7',
    darkGray: '#bdc3c7',
  },
  themeName: Action.DARK_THEME,
};

export default darkTheme;
