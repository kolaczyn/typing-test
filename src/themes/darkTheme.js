export default {
  palette: {
    box: '#222',
    background: '#303030',
    lightGray: 'bisque',
    green: '#55efc4',
    red: '#e17055',
    text: '#fff',
  },
  containerWidth: '768px',
};
