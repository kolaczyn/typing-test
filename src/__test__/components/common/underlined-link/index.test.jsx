import React from 'react';
import { ThemeProvider } from 'styled-components';
import { MemoryRouter } from 'react-router-dom';
import { cleanup, render } from '@testing-library/react';

import UnderlinedLink from '../../../../components/common/underlined-link';
import darkTheme from '../../../../static/themes/darkTheme';

afterEach(cleanup);

describe('<UnderlinedLink />', () => {
  it('renders <UnderlinedLink /> with an icon as <a /> tag', () => {
    const { container } = render(
      <ThemeProvider theme={darkTheme}>
        <UnderlinedLink href="https://reactjs.org/" icon="fas fa-code">
          learn about React
        </UnderlinedLink>
      </ThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders <UnderlinedLink /> without icon as react-router-dom\'s <Link /> tag', () => {
    const { container } = render(
      <MemoryRouter>
        <ThemeProvider theme={darkTheme}>
          <UnderlinedLink to="/log-in">
            log in
          </UnderlinedLink>
        </ThemeProvider>
      </MemoryRouter>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
