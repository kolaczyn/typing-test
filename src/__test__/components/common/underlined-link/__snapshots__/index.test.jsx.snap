// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnderlinedLink /> renders <UnderlinedLink /> with an icon as <a /> tag 1`] = `
<li>
  <a
    class="styles-ja41pm-0 elKeK"
    href="https://reactjs.org/"
    to=""
  >
    <i
      class="fas fa-code"
    />
    <p>
      learn about React
    </p>
  </a>
</li>
`;

exports[`<UnderlinedLink /> renders <UnderlinedLink /> without icon as react-router-dom's <Link /> tag 1`] = `
<li>
  <a
    class="styles-ja41pm-0 cYqeyQ"
    href="/log-in"
  >
    
    <p>
      log in
    </p>
  </a>
</li>
`;
